modules/services/imapnotify.nix

```
@@ -0,0 +1,94 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.imapnotify;
+
+  imapnotifyAccounts =
+    filter (a: a.imapnotify.enable) (attrValues config.accounts.email.accounts);
+
+  genAccountUnit = account: {
+    name = "imapnotify@${account.address}";
+    value = {
+      Unit = {
+        Description = "Execute scripts on IMAP mailbox changes (new/deleted/updated messages) using IDLE for %i";
+        PartOf = [ "network-online.target" ];
+      };
+
+      Service = {
+        ExecStart = "${pkgs.imapnotify}/bin/imapnotify -c %E/imap_inotify/%i-config.js";
```

By "point directly to the file in the Nix store" below I mean to here have something like
```suggestion
        ExecStart = "${pkgs.imapnotify}/bin/imapnotify -c ${genAccountConfig account}";
```
and change `genAccountConfig` to return the result of a `pkgs.writeText` call.

---

I see, I just thought it might be better to use systemd unit interpolation but this works too